### DISCUSSION TITLE
Model ForeignKey field

### DISCUSSION THREAD

**Discussion by ps96068** [jowilf/starlette-admin#605] [Q&A] [answered]
Dear author,

when we have the one-to-many relationship for example we have two simple models:

```
class Post(Base):
    __tablename__ = "posts"

    id: Mapped[int] = mapped_column(primary_key=True, nullable=False)
    .....
    images: Mapped[list["PostImage"]] = relationship(
        "PostImage",
        back_populates="post",
        cascade="all, delete-orphan"
    )

class PostImage(Base):
    __tablename__ = "post_images"

    id: Mapped[int] = mapped_column(primary_key=True, nullable=False)
    post_id: Mapped[int] = mapped_column(ForeignKey("posts.id"), nullable=True)
    .....
    post: Mapped["Post"] = relationship("Post", back_populates="images")
```

when we create ModelView to display them in stralette-admin, we use HasOne and HasMany. EX:

```
class PostView(ModelView):
    fields = [
        IntegerField("id", exclude_from_create=True),
        .....
        HasMany("images", identity="imagini_postare"),
    ]

class PostImageView(ModelView):
    fields = [
        IntegerField("id", exclude_from_create=True),
        ........
        HasOne("post", identity="postari"),
    ]
```

In the dashboard, the value for the relationship fields represents the ids of the relationships.
Can you tell me which method is responsible for displaying the ids and where it is located?
I came across the situation that sometimes it is not relevant to display the ids, sometimes it is more convenient to display for example nikname or another field of the targeted model

Also, if you notice in the above PostImage model we have the "post_id" field. how do I display it in the dashboard? RelationFiled does not work.

**Comment by jowilf** [comment]
https://jowilf.github.io/starlette-admin/user-guide/configurations/modelview/#object-representation

**Reply by ps96068** [reply]
is there any example of usage?

**Reply by jowilf** [reply, marked as answer]
yes https://github.com/jowilf/starlette-admin-demo/blob/8973e1f17a1a2187eb012e6e52bf75290a74489f/app/sqla/models.py#L43
